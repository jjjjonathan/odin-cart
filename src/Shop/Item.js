import React from "react";

export default function Item(props) {
  return (
    <div className="item-card">
      <h4 className="item-title">{props.title}</h4>
      <p className="item-description">{props.description}</p>
      <button
        data-index={props.index}
        type="button"
        onClick={props.onAddToCartClick}
      >
        Add to cart
      </button>
    </div>
  );
}
