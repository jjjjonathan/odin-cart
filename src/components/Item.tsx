import React from 'react';
import CartStatus, {
  AddToCartHandler,
  RemoveFromCartHandler,
} from './CartStatus';

export type ShopItem = {
  title: string;
  description: string;
  price: number;
  quantityInCart: number;
};

type ItemProps = {
  item: ShopItem;
  onAddToCart: AddToCartHandler;
  onRemoveFromCart: RemoveFromCartHandler;
  index: number;
};

const Item = ({ item, onAddToCart, onRemoveFromCart, index }: ItemProps) => {
  return (
    <div className="item-card">
      <h4 className="item-title">{item.title}</h4>
      <p className="item-description">{item.description}</p>
      <p className="price">${item.price}</p>
      <CartStatus
        quantityInCart={item.quantityInCart}
        onAddToCart={onAddToCart}
        onRemoveFromCart={onRemoveFromCart}
        index={index}
      />
    </div>
  );
};

export default Item;
