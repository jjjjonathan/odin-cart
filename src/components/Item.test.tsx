import renderer from 'react-test-renderer';
import Item from './Item';

describe('Item Component', () => {
  it('should match snapshot', () => {
    const item = {
      title: 'Odinfeathers',
      description: 'Faux feathers in pastel gradients',
      price: 595,
      quantityInCart: 0,
    };

    const MockAddToCart = jest.fn();
    const MockRevomeFromCart = jest.fn();

    const tree = renderer
      .create(
        <Item
          item={item}
          onAddToCart={MockAddToCart}
          onRemoveFromCart={MockRevomeFromCart}
          index={0}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
