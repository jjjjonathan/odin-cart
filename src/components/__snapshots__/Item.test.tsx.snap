// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component should match snapshot 1`] = `
<div
  className="item-card"
>
  <h4
    className="item-title"
  >
    Odinfeathers
  </h4>
  <p
    className="item-description"
  >
    Faux feathers in pastel gradients
  </p>
  <p
    className="price"
  >
    $
    595
  </p>
  <div
    className="item-form"
  >
    <form
      data-index={0}
      noValidate={true}
      onSubmit={[Function]}
    >
      <label
        htmlFor="quantity"
      >
        Qty:
      </label>
      <select
        id="quantity"
        name="quantity"
      >
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="3"
        >
          3
        </option>
        <option
          value="4"
        >
          4
        </option>
        <option
          value="5"
        >
          5
        </option>
        <option
          value="6"
        >
          6
        </option>
      </select>
      <button
        type="submit"
      >
        Add to cart
      </button>
    </form>
  </div>
</div>
`;
